hs_core/tests/api/rest/test_resource_list.py

```diff
@@ -166,32 +166,68 @@ def test_resource_list_obsolete(self):
                       msg='obsoleted resource id is not included in returned resource list')
 
     def test_resource_list_by_bounding_box(self):
-        metadata_dict_one = [{'coverage': {'type': 'box', 'value': {'northlimit': '80',
-                                                                    'eastlimit': '40',
-                                                                    'southlimit': '60',
-                                                                    'westlimit': '20',
-                                                                    'units': 'decimal deg'}}}]
-        gen_res_one = resource.create_resource('GenericResource', self.user, 'Resource 1',
-                                               metadata=metadata_dict_one)
-
-        metadata_dict_two = [{'coverage': {'type': 'box', 'value': {'northlimit': '60',
-                                                                    'eastlimit': '110',
-                                                                    'southlimit': '50',
-                                                                    'westlimit': '90',
-                                                                    'units': 'decimal deg'}}}]
-        gen_res_two = resource.create_resource('GenericResource', self.user, 'Resource 2',
-                                               metadata=metadata_dict_two)
-
-        metadata_dict_two = [{'coverage': {'type': 'point', 'value': {'north': '70',
+        metadata_dict_one = [{'coverage': {'type': 'point', 'value': {'north': '70',
                                                                       'east': '70',
                                                                       'units': 'decimal deg'}}}]
-        gen_res_three = resource.create_resource('GenericResource', self.user, 'Resource 2',
-                                                 metadata=metadata_dict_two)
-
-        self.resources_to_delete.append(gen_res_one.short_id)
-        self.resources_to_delete.append(gen_res_two.short_id)
-        self.resources_to_delete.append(gen_res_three.short_id)
-
+        comp_res_one = resource.create_resource('CompositeResource', self.user, 'Resource 1',
+                                                metadata=metadata_dict_one)
+
+        # creating GenericFileMeteData objects now so that when we create a generic aggregation
+        # it's GenericFileMetaData object will have the same id as the id of the resource metadata object
+        if comp_res_one.metadata.id > 1:
+            for _ in range(comp_res_one.metadata.id - 1):
+                GenericFileMetaData.objects.create()
+            GenericFileMetaData.objects.all().delete()
```

Comment:
@hyi I was assuming my comments in line 175-176 would be sufficient to explain the code that follows. Now I realize probably not sufficient.  I will update the comments which may help. 

The purpose of this is to have two Coverage meta objects in two different resources where the `content_object` associated with each of the coverage meta object will have the same id. This id is being used in filtering resources (https://github.com/hydroshare/hydroshare/blob/2280-resource-list-endpoint-fix/hs_core/hydroshare/users.py#L393). When we create a coverage meta object at the resource level, the associated content_object is an instance of CoreMetaData class. When we create a coverage meta object as part of the generic logical file, the associated content_object is an instance of GenericFileMetaData class. So the content_type of the content_object associated with these two coverage meta objects are different. 

Since this modified resource filtering logic (https://github.com/hydroshare/hydroshare/blob/2280-resource-list-endpoint-fix/hs_core/hydroshare/users.py#L393)  now uses both content_object id and content_type id, I wanted to test a case where I have 2 resources that have coverage meta with same content_object id but different content_type. 

Since the value for the 'id' (used in line 177) is auto generated (sequential number generation) by the database, we can't set it to a value that we want. So in order to have a GenericFileMataData object with id equal to `comp_res_one.metadata.id`, here we are first creating a number of GenericFileMetaData objects and then deleting them. Then when we create a GenericLogicalFile object (line 191) it creates a GenericFileMetaData object which will have the the id same as the `comp_res_one.metadata.id` 

If  `comp_res_one.metadata.id` is equal to 5, we have to create 4 GenericFileMataData objects and then delete them. Then when we create GenericFileMataData object as part of creating generic logical file the id assigned to GenericFileMataData object by the DB would be 5.

You may be wondering about the test case where I have 2 resources that have coverage meta with different content_object id but same content_type. This test case is also covered. When we create 2 composite resources and add resource level coverage meta to both these resources we meet this test case.